### DISCUSSION TITLE
SimpleMap in move

### DISCUSSION THREAD

**Discussion by Evan-dev999** [aptos-labs/aptos-developer-discussions#176] [Questions] [answered]
### Describe your question in detail.

Is there anyway to pass a variable type SimpleMap to a function. I can't do that but i saw an old source code that still can do that.

### What error, if any, are you getting?

type `&simple_map::SimpleMap<string::String, string::String>` is not supported as a parameter type

### What have you tried or looked at?  Or how can we reproduce the error?

I try to compile my code and face with the above error. You can reproduce the error by compile a smart contract that have a function with parameter type SimpleMap

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment, marked as answer]
It might be because you're using an entry function. Entry functions cannot take structs as params, only primitive types (and special cases like `Object` and `String`). `SimpleMap` is a [struct](https://github.com/aptos-labs/aptos-core/blob/03858b9a393ab3076b23c5172124b1c4fbb126f1/aptos-move/framework/aptos-stdlib/sources/simple_map.move#L17).

Refer to the [docs](https://aptos.dev/move/book/functions/#entry-modifier) on entry functions for more information.

**Reply by kshitijc1506** [reply]
Also, be aware that the feature allowing structs as parameters for entry functions may be added in the future, so verify whether this is available at the time of reading.
